### ISSUE TITLE
Determine best methodology for normalizing and augmenting training audio

### ISSUE THREAD

**Issue by acmiyaguchi** [dsgt-birdclef/birdclef-2023#31]
There's 190 hours of training audio in the dataset, but the distribution of the audio is not even across species nor across individual tracks. There are audio tracks that are 5 seconds long, and tracks that are 10 minutes long. We should try to decide on what constitutes a good size for training/testing data.

We can build off of some design constraints to simply the problem:

- Hidden test data (leaderboard test dataset) is a 60 second audio clip
- BirdNET generates embeddings on 3 second windows
- BirdNET embeddings are 320-dim double vector
- Sliding window for sequential embeddings should probably be between 1-3 seconds
- Context window on the transformer model should probably be as large as we can afford (a telsa T4 gpu gives us 16gb vram, we should aim for a batch size of 4 to 16 during dev, and consider a more powerful configuration during training).

How do we handle some of the following cases:

- A very short clip (under 5 seconds)
- A very long clip (over 2 minutes)

We have interesting tools at our disposal. When we perform sound separation, we often obtain a background track that has no birdcalls. Could we synthesize relatively good training data using separated audio that's mixed in plausible ways? 

The goal of this issue is to create a series of notebooks that investigates the problem, and proposes an algorithm that we can implement for training. Note caveats. Compare and constrast approaches to normalizing audio clips so that they are 15, 30, and 60 seconds in length. 

It would be useful to specify a useful web interface to explore the data.

**Comment by acmiyaguchi** [comment]
We're doing 30 second clips more or less, and pad the data so its divisble by 3. This works good more or less, which augmentation coming from noise and mixit.
